sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v2/query.sql

```diff
@@ -0,0 +1,291 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    category AS event_category,
+    name AS event_name,
+    timestamp AS event_timestamp,
+    client_info,
+    metadata,
+    normalized_os,
+    normalized_os_version,
+    normalized_country_code,
+    normalized_channel,
+    ping_info,
+    extra AS event_details,
+    metrics
+  FROM
+    -- https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/pings/newtab
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad', 'pocket')
+    AND name IN ('closed', 'opened', 'impression', 'issued', 'click', 'save', 'topic_click')
+),
+visit_metadata AS (
+  SELECT
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+    submission_date,
+    ANY_VALUE(client_info.client_id) AS client_id,
+    ANY_VALUE(metrics.uuid.legacy_telemetry_client_id) AS legacy_telemetry_client_id,
+    ANY_VALUE(normalized_os) AS normalized_os,
+    ANY_VALUE(normalized_os_version) AS normalized_os_version,
+    ANY_VALUE(normalized_country_code) AS country_code,
+    ANY_VALUE(normalized_channel) AS channel,
+    ANY_VALUE(client_info.app_display_version) AS browser_version,
+    "Firefox Desktop" AS browser_name,
+    ANY_VALUE(metrics.string.search_engine_default_engine_id) AS default_search_engine,
+    ANY_VALUE(metrics.string.search_engine_private_engine_id) AS default_private_search_engine,
+    ANY_VALUE(metrics.boolean.pocket_is_signed_in) AS pocket_is_signed_in,
+    ANY_VALUE(metrics.boolean.pocket_enabled) AS pocket_enabled,
+    ANY_VALUE(metrics.boolean.pocket_sponsored_stories_enabled) AS pocket_sponsored_stories_enabled,
+    ANY_VALUE(metrics.boolean.topsites_enabled) AS topsites_enabled,
+    ANY_VALUE(metrics.string.newtab_homepage_category) AS newtab_homepage_category,
+    ANY_VALUE(metrics.string.newtab_newtab_category) AS newtab_newtab_category,
+    ANY_VALUE(metrics.boolean.newtab_search_enabled) AS newtab_search_enabled,
+    ANY_VALUE(metrics.quantity.topsites_rows) AS topsites_rows,
+    ANY_VALUE(metrics.string_list.newtab_blocked_sponsors) AS newtab_blocked_sponsors,
+    ANY_VALUE(ping_info.experiments) AS experiments,
+    MIN(IF(event_name = "opened", event_timestamp, NULL)) AS newtab_visit_started_at,
+    MIN(IF(event_name = "closed", event_timestamp, NULL)) AS newtab_visit_ended_at,
+    ANY_VALUE(
+      IF(event_name = "opened", mozfun.map.get_key(event_details, "source"), NULL)
+    ) AS newtab_open_source,
+    LOGICAL_OR(
+      event_name = "click" OR
+      event_name = "issued" OR
+      event_name = "save"
+    ) as had_non_impression_engagement
+    FROM
+      events_unnested
+  GROUP BY
+    newtab_visit_id,
+    submission_date
+  HAVING
+    newtab_visit_id IS NOT NULL
+),
+search_events AS (
```

Comment:
The pattern here is fairly straightforward for search, topsites, and pocket. 2 CTE's each. The first groups by the context they have (search engine and search access point in the search case for e.g.) and COUNTIF's for the metrics. The second creates an array of `x_interactions` grouped by newtab_visit_id.